### PR TITLE
Edit & delete My Posts | Delete Comments | List only my posts

### PR DESCRIPTION
**Pull request by jazwu**
## Description

- modified post_detail.html: modified UI, added Edit and Delete buttons and delete confirmation modals
- added post_edit.html
- modified community_home.html: modified UI, added "write a post" button, added a dropdown button to either see all posts or just my posts
- added text_filters template tags for post content display on the community home page

## How did you test the changes?
**1. create a post**
image
click "Write a post" button and it will direct to the post creation page
**2. edit my post**
only my posts show edit buttons
image
click the Edit button, it will direct to edit your post page
image
after modifying the title and contents, publish your post again
image
you'll see the post has been modified and there is also a "edited" mark showing under the title
**3. delete my post**
only my posts show delete buttons
when clicking on the delete button, it will prompt a deletion confirmation modal
image
when click Confirm Delete, the post will be deleted and it redirects to the community home
**4. delete comments**
only my comments or the comments under my posts will show delete buttons
image
when clicking on the delete button, it will prompt a deletion confirmation modal
image
after clicking "comfirm delete", the comment will be deleted and back to view_post page
image
**5. see all posts or just my posts**
default or select "all the posts", it displays all the posts
image
select "my posts", it displays only mine
image

### CODE REVIEW

File: healthScore/templates/header.html [hunk 1]
```diff
@@ -93,11 +93,12 @@
               
               {% if user.is_authenticated and user.is_patient %}
                 <ul class="nav col-12 col-lg-auto mb-2 justify-content-center mb-md-0">
-                    <li><a href="/" class="nav-link">Home</a></li>
+                    <!-- <li><a href="/" class="nav-link">Home</a></li> -->
```
Comment (sarthakgoel1997) [inline healthScore/templates/header.html]: Is this change intended?

File: healthScore/templatetags/__init__.py
Comment (sarthakgoel1997) [inline healthScore/templatetags/__init__.py:1]: Please remove this file if it is not required

**Review by sarthakgoel1997 — CHANGES_REQUESTED** [review]

File: healthScore/templates/post_details.html [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+        <ul class="comment-container">
+            {% if comments %}
+            {% for comment in comments %}
+            <li class="comment">
+                <div class="icon-bar">
+                    <div>
+                        <h6>{{ comment.commenter }}</h6>
+                    </div>
+                    {% if user.is_authenticated %}
+                    {% if user == comment.commenter or user == comment.post.user %}
```
Comment (shreyas2499) [inline healthScore/templates/post_details.html:149]: Could we club these two conditions and write it like this:

if user.is_authenticated and (user == comment.commenter or user == comment.post.user) ?

File: healthScore/templates/header.html (hunk 1 above)
Comment (jazwu) [reply healthScore/templates/header.html]: i'll uncomment it

File: healthScore/templatetags/__init__.py
Comment (jazwu) [reply healthScore/templatetags/__init__.py:1]: this is required because it is a package that will be loaded to HTML pages otherwise you cannot load it @sarthakgoel1997

File: healthScore/templates/post_details.html (hunk 2 above)
Comment (jazwu) [reply healthScore/templates/post_details.html:149]: i tried but it didnt work, that's why @shreyas2499

Comment (shreyas2499) [reply healthScore/templates/post_details.html:149]: Ohh. Alright then

**Review by shreyas2499 — APPROVED** [review]

**Review by sarthakgoel1997 — APPROVED** [review]
